src/Adapter/Order/AbstractOrderHandler.php

```diff
@@ -325,7 +326,10 @@ protected function getProductRegularPrice(
             false,
             $order->id_customer, // We still use the customer ID in case this customer has some special prices
             null, // But we keep the cart null as we don't want this order overridden price
-            $order->{Configuration::get('PS_TAX_ADDRESS_TYPE', null, null, $order->id_shop)}
+            $order->{Configuration::get('PS_TAX_ADDRESS_TYPE', null, null, $order->id_shop)},
+            $specificPriceOutput,
+            true,
+            false
```

Comment:
Maybe add a comment to indicate we explicitly ask for the price without group reduction?

Although I'm worried about this one because if we want to add a product that has a group reduction then we can't do it by default, or we can but it would be considered as a specific price

Reply:
This method `getProductRegularPrice` is only used in `updateSpecificPrice` which is called in `AddProductToOrderHandler` and `UpdateProductInOrderHandler`
When we add or update a product in order, the prices are given in parameter, and they are taken as they are, so if the product has group reduction, it has to be included in the price given

You are right, it will be in all cases saved as a specific price

Reply:
For the product add for example, if the interface is used, the user will hit SearchProductHandler::getProductPriceForOrder to get the product details he wants. This method will return the effective product price, group reduction included